conjur_api/providers/ldap_authentication_strategy.py

```diff
@@ -0,0 +1,58 @@
+"""
+LdapAuthenticationStrategy module
+
+This module holds the LdapAuthenticationStrategy class
+"""
+
+from conjur_api.http.endpoints import ConjurEndpoint
+from conjur_api.interface.credentials_store_interface import CredentialsProviderInterface
+from conjur_api.providers.authn_authentication_strategy import AuthnAuthenticationStrategy
+from conjur_api.wrappers.http_wrapper import HttpVerb, invoke_endpoint
+
+# pylint: disable=too-few-public-methods
+class LdapAuthenticationStrategy(AuthnAuthenticationStrategy):
+    """
+    LdapAuthenticationStrategy class
+
+    This class implements the "ldap" strategy of authentication. This is almost the same as
+    AuthnAuthenticationStrategy, except that it has an additional parameter "service_id" which
+    is included in the authentication request url.
+    """
+
+    def __init__(
+            self,
+            credentials_provider: CredentialsProviderInterface,
+            service_id: str
```

Comment:
Is there a way  we could pass `service_id` as an `authenticate` parameter instead of as an instance variable? Seems like a candidate for a `ConjurConnectionInfo` member instead.

This might not be an important use-case for the Python API, but consider:
- One `AuthnAuthenticationStrategy` instance can auth to any target (URL/account/username combo) based on parameters passed to `authenticate`
- One `LdapAuthenticationStrategy` instance can auth to any target based on `authenticate` parameters, given that the target has an LDAP authenticator named `service_id`

Reply:
It's a good point. I considered adding service_id to ConjurConnectionInfo but it seemed to me that since this is something specific to the ldap authenticator that it was best not to introduce it into the common code used by all authn strategies. I can definitely see an argument the other way as well though.

Reply:
Another point to consider: authn-jwt, azure, and aws iam also has a service_id, but authn-gcp doesn't. Does the fact that multiple auth strategies use a service_id justify making it part of the shared code?

Reply:
I added a new commit, db0b72607fc34ffa3f0db37deba6af6fb1fa91c5, that moves the service id to `ConjurConnectionInfo`. If we like it I'll squash it into the previous commit.

Reply:
Do you need to update the README.md with this?

Reply:
Yes, thanks for catching that

Reply:
Updated